src/main.c

```diff
@@ -27,14 +27,42 @@ gboolean info_noverify, info_dumpcert = FALSE;
 gboolean status_detailed = FALSE;
 gchar *output_format = NULL;
 
+#include <sys/ioctl.h>
+#include <stdio.h>
+#include <unistd.h>
+
+static gchar* get_progress(gint percentage)
+{
+	struct winsize w;
+	GString *printbuf = NULL;
+	gint pbar_len = 0;
+	
+	g_return_val_if_fail(percentage <= 100, NULL);
+
+	/* obtain terminal window parameters */
+	ioctl(STDOUT_FILENO, TIOCGWINSZ, &w);
+	pbar_len = w.ws_col - 1 - 1 - 5;
```

Comment:
Maybe limit this to 100? We don't have more information anyway.

Reply:
Neither bitbake nor wget etc. limit the width of the progress bar. Thus I would prefer using the full width here, too.

Reply:
Hmm, OK.